frontend/src/components/SortOptionDropDown.tsx

```diff
@@ -0,0 +1,54 @@
+import { DropdownMenuContent } from '@radix-ui/react-dropdown-menu';
+import { Button } from './ui/button';
+import {
+  DropdownMenu,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from './ui/dropdown-menu';
+
+type Props = {
+  onChange(value: string): void;
+  sortOption: string;
+};
+
+const SORT_OPTIONS = [
+  {
+    label: 'Best Match',
+    value: 'bestMatch',
+  },
+  {
+    label: 'Delivery price',
+    value: 'deliveryPrice',
+  },
+  {
+    label: 'Estimated delivery time',
+    value: 'estimatedDeliveryTime',
+  },
+];
+
+const SortOptionDropDown = ({ onChange, sortOption }: Props) => {
+  const selectedSortOption =
+    SORT_OPTIONS.find((option) => option.value === sortOption)?.label ||
+    'Best Match';
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger className="cursor-pointer">
+        <Button variant="outline" className="w-full">
+          Sort by : {selectedSortOption}
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent>
+        {SORT_OPTIONS.map((option) => (
+          <DropdownMenuItem
+            className="cursor-pointer"
+            onClick={() => onChange(option.value)}
+          >
+            {option.label}
+          </DropdownMenuItem>
+        ))}
+      </DropdownMenuContent>
+    </DropdownMenu>
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/42#issuecomment-2407957979>#5238e7</a><span> - 10/11/2024, 06:53 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The component lacks proper keyboard navigation support, which is crucial for accessibility.
</div>
<div id="fix">
<b>Fix:</b> Implement keyboard navigation support using the Radix UI Dropdown Menu component's built-in accessibility features.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -33,20 +33,22 @@
    return (
      <DropdownMenu>
        <DropdownMenuTrigger className="cursor-pointer">
 -        <Button variant="outline" className="w-full">
 +        <Button variant="outline" className="w-full" aria-haspopup="true">
            Sort by : {selectedSortOption}
          </Button>
        </DropdownMenuTrigger>
        <DropdownMenuContent>
          {SORT_OPTIONS.map((option) => (
            <DropdownMenuItem
 +            key={option.value}
              className="cursor-pointer"
              onClick={() => onChange(option.value)}
 +            onSelect={() => onChange(option.value)}
            >
              {option.label}
            </DropdownMenuItem>
          ))}
        </DropdownMenuContent>
      </DropdownMenu>
    );
```
</div>
</details>
</div>
<br>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The component lacks proper ARIA attributes for improved accessibility, particularly for screen readers.
</div>
<div id="fix">
<b>Fix:</b> Add appropriate ARIA attributes to the DropdownMenu components to improve accessibility.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -31,20 +31,22 @@
      SORT_OPTIONS.find((option) => option.value === sortOption)?.label ||
      'Best Match';
    return (
 -    <DropdownMenu>
 -      <DropdownMenuTrigger className="cursor-pointer">
 +    <DropdownMenu>
 +      <DropdownMenuTrigger className="cursor-pointer" aria-haspopup="true" aria-expanded="false">
          <Button variant="outline" className="w-full">
            Sort by : {selectedSortOption}
          </Button>
        </DropdownMenuTrigger>
 -      <DropdownMenuContent>
 +      <DropdownMenuContent role="menu">
          {SORT_OPTIONS.map((option) => (
            <DropdownMenuItem
              className="cursor-pointer"
              onClick={() => onChange(option.value)}
 +            role="menuitem"
 +            key={option.value}
            >
              {option.label}
            </DropdownMenuItem>
          ))}
        </DropdownMenuContent>
      </DropdownMenu>
    );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged